test/smoke/dashCore.js

```diff
@@ -1,18 +1,79 @@
-describe('DashCore', () => {
-  describe('All nodes', () => {
-    // `subversion` from `getNetworkInfo` response.
-    it('should have correct network type');
+const RpcClient = require('@dashevo/dashd-rpc/promise');
+const SSH = require('simple-ssh');
+const fs = require('fs');
+const utils = require('../testUtils');
 
-    // The number of blocks should be almost the same (-3/+3) and block hash of particular
-    // block height should be the same. Using `blocks` and `bestblockhash` from `GetBlockChainInfo`.
-    it('should propagate blocks');
-  });
-  describe('Masternodes', () => {
-    // masternode status
-    it('should be masternodes');
-  });
-  describe('Miners', () => {
-    // waitfornewblock
-    it('should mine blocks');
+const getNetworkConfig = require('../../lib/test/getNetworkConfig');
+
+const networkConfig = getNetworkConfig();
+
+
+networkConfig.inventory.masternodes.hosts.forEach((nodeName) => {
+  describe(`DashCore ${nodeName}`, () => {
+    const config = {
+      protocol: 'http',
+      user: 'dashrpc',
+      pass: 'password',
+      host: networkConfig.inventory._meta.hostvars[nodeName].public_ip,
+      port: 20002,
+    };
+    const rpc = new RpcClient(config);
+
+    describe('All nodes', () => {
+      it('should have correct network type/subversion from `getNetworkInfo`', async () => {
+        const blockHash = await rpc.getNetworkInfo();
+        const masternodelist = await rpc.masternodelist();
+        const version = masternodelist.result[Object.keys(masternodelist.result)[0]].daemonversion;
+        expect(blockHash.error).to.be.equal(null);
+        expect(blockHash.result.networkactive).to.be.equal(true);
+        expect(blockHash.result.subversion).to.be
+          .equal(`/Dash Core:${version}(${networkConfig.network.type}=${networkConfig.network.name})/`);
+      });
+
+      // The number of blocks should be almost the same (-3/+3) and block hash of particular
+      // block height should be the same. Using `blocks`
+      // and `bestblockhash` from `GetBlockChainInfo`.
+      it('should have correct blockhash and blocks count', async () => {
+        const bestBlockHash = await rpc.getBestBlockHash();
```

Comment:
Why do we compare RPC response with cli? I don't understand the meaning. The idea was to compare values from all nodes and the average should be the same.

Reply:
fixed. now compare only rpc responses between all nodes.